include/picongpu/plugins/openPMD/GetComponentsType.hpp

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2021 Sergei Bastrakov
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <pmacc/traits/GetComponentsType.hpp>
+
+#include <type_traits>
+
+namespace pmacc
+{
+    namespace traits
+    {
+        /** Get component type trait for bools in openPMD output
+         *
+         * ADIOS2 backend of openPMD API currently does not support bool datasets #3732.
+         * So with this specialization, PIConGPU particle attributes of type bool (e.g. radiationMask,
+         * transitionRadiationMask) are treated as chars.
+         *
+         * This requires sizeof(bool) == sizeof(char), ::type is defined only in this case.
+         */
+        template<>
+        struct GetComponentsType<bool>

```

Comment:
Could you maybe add a little comment which file contains the original definition of `GetComponentsType` (`pmacc/traits/GetComponentsType.hpp`)? It's was not immediately obvious to me that this "overrides" a definition found somewhere completely different.

Reply:
In this case it's actually the "standard" location that corresponds to the namespace structure. But sure, I can add a sentence about it.

Reply:
Done